Build failure: azahar

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- run `nix flake update`
- run `sudo nixos-rebuild switch --flake .`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310525038

### Relevant log output

```console
error: builder for '/nix/store/zvyjil8jv52rh60am3pqxs4313isb74z-azahar-2123.3.drv' failed with exit code 2;
       last 25 log lines:
       > [ 90%] Automatic RCC for ../../dist/qt_themes/colorful_dark/theme_colorful_dark.qrc
       > Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
       > Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
       > If this causes problems, reconfigure your locale. See the locale(1) manual
       > for more information.
       > [ 90%] Automatic RCC for ../../dist/compatibility_list/compatibility_list.qrc
       > [ 90%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/cmake_pch.hxx.gch
       > [ 90%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/qrc_languages.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/citra_qt_autogen/mocs_compilation.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/aboutdialog.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/applets/mii_selector.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/applets/swkbd.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/bootmanager.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/compatibility_list.cpp.o
       > [ 92%] Building CXX object src/citra_qt/CMakeFiles/citra_qt.dir/camera/camera_util.cpp.o
       > /build/source/src/citra_qt/bootmanager.cpp:42:10: fatal error: qpa/qplatformnativeinterface.h: No such file or directory
       >    42 | #include <qpa/qplatformnativeinterface.h>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > make[2]: *** [src/citra_qt/CMakeFiles/citra_qt.dir/build.make:667: src/citra_qt/CMakeFiles/citra_qt.dir/bootmanager.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:2693: src/citra_qt/CMakeFiles/citra_qt.dir/all] Error 2
       > make[1]: *** Waiting for unfinished jobs....
       > [ 92%] Built target tests
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/zvyjil8jv52rh60am3pqxs4313isb74z-azahar-2123.3.drv'.
error: 1 dependencies of derivation '/nix/store/jc2gmr8b6j4xdlha18npysw9da9y2j36-azahar-2123.3-fish-completions.drv' failed to build
```

### Additional context

This is probably an issue with Qt6, because I can replicate these logs for melonDS too

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.3, NixOS, 25.11 (Xantusia), 25.11.20251019.5e2a59a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/sunny/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/sunny/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/sunny/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/sunny/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/zxl9ivn2v7xcrzlf4j036fl6av34zwv9-lix-2.93.3/share`
 - nixpkgs: `/nix/store/7agp54mgffm9m1wc1kgmkm37pvy18qhf-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/az/azahar/package.nix
+++ b/pkgs/by-name/az/azahar/package.nix
@@ -44,6 +44,7 @@
   nix-update-script,
   darwinMinVersionHook,
   apple-sdk_12,
+  fetchpatch2,
 }:
 let
   inherit (lib)
@@ -65,6 +66,12 @@ stdenv.mkDerivation (finalAttrs: {
   patches = [
     # https://github.com/azahar-emu/azahar/pull/1305
     ./fix-zstd-seekable-include.patch
+
+    # TODO: Remove in next release
+    (fetchpatch2 {
+      url = "https://github.com/azahar-emu/azahar/commit/1f483e1d335374482845d0325ac8b13af3162c53.patch?full_index=1";
+      hash = "sha256-9rmRbv7VFMhHly5qTGaeBLpvtWMu6HkCGUUM+t78Meg=";
+    })
   ];
 
   strictDeps = true;

